packages/jupyterlab-manager/src/manager.ts

```diff
@@ -113,16 +113,20 @@ class WidgetManager extends ManagerBase<Widget> implements IDisposable {
       await this.handle_comm_open(oldComm, msg);
     };
 
-    context.session.kernelChanged.connect((sender, args) => {
+    context.sessionContext.kernelChanged.connect((sender, args) => {
       this._handleKernelChanged(args);
     });
 
-    context.session.statusChanged.connect((sender, args) => {
+    context.sessionContext.statusChanged.connect((sender, args) => {
       this._handleKernelStatusChange(args);
     });
+    context.sessionContext.connectionStatusChanged.connect((sender, args) => {
+      this._handleKernelConnectionStatusChange(args);
+    });
+
 
-    if (context.session.kernel) {
-      this._handleKernelChanged({oldValue: null, newValue: context.session.kernel});
+    if (context.sessionContext.session?.kernel) {
```

Comment:
Cool, I did not know this syntax.

Reply:
New in TS 3.7: https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining